eng/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -20,23 +23,31 @@ jobs:
     - ${{ each config in parameters.MatrixConfigs }}:
       - ${{ if eq(config.GenerateVMJobs, 'true') }}:
         - pwsh: |
-            Write-Host "DISPLAY NAME FILTER: $(displayNameFilter)"
+            $extraFilters = "${{ join('","', parameters.MatrixFilters) }}"
+            $filters = ,"container=^$","SupportedClouds=^$|${{ parameters.CloudConfig.Cloud }}"
+            if ($extraFilters) { $filters = @($extraFilters) + $filters }
+            Write-Host "DISPLAY NAME FILTER: `"$(displayNameFilter)`"; FILTERS: `"$filters`""
             eng/scripts/job-matrix/Create-JobMatrix.ps1 `
               -ConfigPath ${{ config.Path }} `
               -Selection ${{ config.Selection }} `
               -DisplayNameFilter "$(displayNameFilter)" `
-              -Filters @("container=^$", "SupportedClouds=^$|${{ parameters.CloudConfig.Cloud }}")
+              -Filters $filters `
+              -NonSparseParameters "${{ join('","', config.NonSparseParameters) }}"
           name: generate_vm_job_matrix_${{ config.Name }}
           displayName: Generate VM Job Matrix
 
       - ${{ if eq(config.GenerateContainerJobs, 'true') }}:
         - pwsh: |
-            Write-Host "DISPLAY NAME FILTER: $(displayNameFilter)"
+            $extraFilters = "${{ join('","', parameters.MatrixFilters) }}"
+            $filters = ,"container=.*","SupportedClouds=^$|${{ parameters.CloudConfig.Cloud }}"
+            if ($extraFilters) { $filters = @($extraFilters) + $filters }
+            Write-Host "DISPLAY NAME FILTER: `"$(displayNameFilter)`"; FILTERS: `"$filters`""
             eng/scripts/job-matrix/Create-JobMatrix.ps1 `
               -ConfigPath ${{ config.Path }} `
               -Selection ${{ config.Selection }} `
               -DisplayNameFilter "$(displayNameFilter)" `
-              -Filters @("container=.*", "SupportedClouds=$^$|{{ parameters.CloudConfig.Cloud }}")
+              -Filters $filters `
```

Comment:
Given you filter out empty/null filters couldn't you just do:
`@("container=.*", "SupportedClouds=$^$|{{ parameters.CloudConfig.Cloud }}", "${{ join('","', parameters.MatrixFilters) }}")` 

Assuming we wanted to keep this on one line.